spec/requests/application_controller_spec.rb

```diff
@@ -33,4 +33,67 @@
     end
   end
 
+  describe "#handle_theme" do
+    let(:theme) { Theme.create!(user_id: -1, name: 'bob', user_selectable: true) }
+    let(:theme2) { Theme.create!(user_id: -1, name: 'bobbob', user_selectable: true) }
+    let(:user) { Fabricate(:user) }
+    let(:admin) { Fabricate(:admin) }
+
+    before do
+      sign_in(user)
+    end
+
+    it "selects the theme the user has selected" do
+      user.user_option.update_columns(theme_ids: [theme.id])
+
+      get "/"
+      expect(controller.theme_ids).to eq([theme.id])
```

Comment:
We need to test for the `response.code` whenever we make a request as it makes it easier to debug should the test fails